bin/validate-markdown.ts

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env node
+
+/**
+ * Validates that markdown files exist for all HTML pages in the public directory.
+ * This script ensures the markdown generation process completed successfully.
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import fastGlob from 'fast-glob';
+
+const publicDir = path.join(process.cwd(), 'public', 'docs');
+
+// Constants for content validation (must match generateMarkdown.ts)
+const REDIRECT_PAGE_MAX_SIZE = 1000; // Maximum size in bytes for redirect pages
+
+interface ValidationResult {
+  totalPages: number;
+  markdownFound: number;
+  markdownMissing: number;
+  redirectPages: number;
+  missingFiles: string[];
+}
+
+const validateMarkdownFiles = async (): Promise<ValidationResult> => {
+  // Find all index.html files in the docs directory
+  const htmlFiles = await fastGlob('**/index.html', {
+    cwd: publicDir,
+    absolute: false,
+  });
+
+  const result: ValidationResult = {
+    totalPages: htmlFiles.length,
+    markdownFound: 0,
+    markdownMissing: 0,
+    redirectPages: 0,
+    missingFiles: [],
+  };
+
+  for (const htmlFile of htmlFiles) {
+    // Get the directory of the HTML file
+    const dir = path.dirname(htmlFile);
+
+    // Check if this is a redirect page (skip validation for these)
+    const htmlPath = path.join(publicDir, htmlFile);
+    const htmlContent = fs.readFileSync(htmlPath, 'utf8');
+
+    if (htmlContent.length < REDIRECT_PAGE_MAX_SIZE && htmlContent.includes('window.location.href')) {
```

Comment:
This seems flimsy to me. Looking at the generated html, we have a pretty consistent case for this - something like 
```
<script>window.location.href="/docs/auth"</script>
```
We can look for this directly.